server/datastore/mysql/software.go

```diff
@@ -2076,7 +2085,7 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 		return nil, nil, ctxerr.Wrap(ctx, err, "build named query for list host software")
 	}
 
-	if opts.IncludeAvailableForInstall {
+	if opts.IncludeAvailableForInstall && !opts.VulnerableOnly {
```

Comment:
They are mutually exclusive, since installers don't have vulnerabilities. 

> Is there a product specification for the order of precedence
I don't think so, at least not on the ticket.

Reply:
Ok, maybe we should create a follow up ticket for product document the mutually exclusive filters and also for us to take a closer look if there's any other places where we want to be enforcing this (for example, what does a user see if they try navigating directly to the host details page with the mutually exclusive filters set in the URL).